packages/@react-aria/datepicker/src/useDateSegment.ts

```diff
@@ -44,7 +44,7 @@ export function useDateSegment<T extends DateValue>(props: DatePickerProps<T> &
 
   if (segment.type === 'month') {
     let monthTextValue = monthDateFormatter.format(state.dateValue);
-    textValue = monthTextValue !== textValue ? `${textValue} - ${monthTextValue}` : monthTextValue;
+    textValue = monthTextValue !== textValue ? `${textValue} – ${monthTextValue}` : monthTextValue;
```

Comment:
This avoids VoiceOver reading "minus" for the dash.

Reply:
en dash

Reply:
No, it's a unicode character so the screen reader will interpret it in whatever language it needs to